### PR TITLE
Get/set commit sha on deploy

### PR DESCRIPTION
**Pull request by jkburges**
In the pre-command hook, i.e. before deploy, fetch previous deploy's git sha and export it to an env var `LAST_SUCCESSFUL_COMMIT`.

In post-command hook, write it. This allows more reliable checking on whether or not a deploy needs to happen.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 steps:
   - label: "Deploy"
-    command: "echo 'just testing myself'"
+    command: "echo 'just testing myself', last successful commit: $$LAST_SUCCESSFUL_COMMIT"
```
Comment (Skwai) [inline .buildkite/pipeline.yml:3]: Is this interpolation the correct syntax? Should it not be `${LAST_SUCCESSFUL_COMMIT}`?

Comment (jkburges) [reply .buildkite/pipeline.yml:3]: It's correct: it stops the variable between interpolated at the beginning of pipeline execution, rather, it is the remains as a variable in the command, i.e. `echo ... $LAST_SUCCESSFUL_COMMIT` at command run time.

Doc: https://buildkite.com/docs/agent/v3/cli-pipeline#environment-variable-substitution

Comment (jkburges) [reply .buildkite/pipeline.yml:3]: Sharp eyes though 👀

**Comment by jkburges** [comment]
There must be a bug looking up author, see in slack is says "Author: <@>" instead of your slack handle. I'll take a look.

**Comment by jkburges** [comment]
> There must be a bug looking up author, see in slack is says "Author: <@>" instead of your slack handle. I'll take a look.

Hmmm it's because:

```
BUILDKITE_BUILD_AUTHOR_EMAIL="cj...@gmail.com"
```

I was under the impression that the buildkite user's email would be used but clearly not.

**Comment by jkburges** [comment]
@Skwai is there an option in https://buildkite.com/user/settings to set your primary email, and is it set to your gmail or biteable address?

**Comment by Skwai** [comment]
FYI Added the standard branch protection stuff (to require approval before merge allowed)

**Comment by Skwai** [comment]
image


> @Skwai is there an option in https://buildkite.com/user/emails to set your primary email, and is it set to your gmail or biteable address?

image

**Comment by Skwai** [comment]
I just deleted my personal email from the account, hopefully that helps?

**Comment by jkburges** [comment]
> I just deleted my personal email from the account, hopefully that helps?

Hmm I was under the impression that the `BUILDKITE_BUILD_AUTHOR_EMAIL` would be set to the user's primary email address, and I think maybe also any other emails that are used in git commits need to be in buildkite so it has some way of mapping the commit author email to a buildkite user.

But that's what you had there, however your gmail address wasn't verified - so maybe that was the issue?

I'll do some testing and get back to you.

**Comment by jkburges** [comment]
> But that's what you had there, however your gmail address wasn't verified - so maybe that was the issue?

I think this might be it, I used my gmail address, unlinked it from bk - same behaviour as for you initially. Added it to bk, but left unverified. Also the same. Once I verified it, things started to work.

**Comment by Skwai** [comment]
Weird... you'd think it would use the verified one.

**Comment by jkburges** [comment]
> Weird... you'd think it would use the verified one.

It has to get from the git commit author (the gmail address in your case) though

**Comment by Skwai** [comment]
Ah I get you

**Review by Skwai — APPROVED** [review]

**Pull request merged**